roslibrust_codegen/src/utils.rs

```diff
@@ -129,3 +149,58 @@ fn message_files_from_path(path: &Path, ext: &str) -> io::Result<Vec<PathBuf>> {
 
     Ok(msg_files)
 }
+
+fn determine_ros_version(
+    path: impl AsRef<Path> + std::fmt::Debug,
+) -> Result<RosVersion, std::io::Error> {
+    use std::fs::File;
+    use std::io::BufReader;
+    use xml::reader::{EventReader, ParserConfig, XmlEvent};
+    const TAG: &'static str = "buildtool_depend";
+
+    let file = File::open(&path)?;
+    let reader = BufReader::new(file);
+    let parser = EventReader::new_with_config(
+        reader,
+        ParserConfig {
+            trim_whitespace: true,
+            ignore_comments: true,
+            ..Default::default()
+        },
+    );
+
+    let mut in_tag = false;
+    for e in parser {
+        match e {
+            Ok(XmlEvent::StartElement { name, .. }) => {
+                if name.local_name == TAG {
+                    in_tag = true;
+                }
+            }
+            Ok(XmlEvent::Characters(data)) => {
+                if in_tag {
+                    log::debug!("Got data inside of {TAG}: {data}");
+                    match data.as_str() {
+                        "catkin" => {
+                            return Ok(RosVersion::ROS1);
+                        }
+                        "ament_cmake" => {
```

Comment:
Is it also possible for this to be colcon or does this imply colcon? (not actually sure of the relationship between these tools)

Reply:
I asked this earlier today:
https://answers.ros.org/question/410017/how-to-determine-if-a-package-is-ros1-or-ros2/

We'll see if I can get and answer, I did the best digging I could through the ROS documentation and couldn't find a better answer. As far as I can tell this is never colcon, and instead colcon -> ament is the same relations ship as `catkin build` to `catkin_make`. I think colcon just invokes ament for you...

I'm not very sure about any of this hence the failure path in the code and the question...